lib/cmsis_rtos_v2/kernel.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <kernel_structs.h>
+#include <ksched.h>
+#include <cmsis_os2.h>
+
+extern u32_t _tick_get_32(void);
+
+/**
+ * @brief Get RTOS Kernel Information.
+ */
+osStatus_t osKernelGetInfo(osVersion_t *version, char *id_buf, uint32_t id_size)
+{
+	char os_str[] = "Zephyr VMM.mm.pp";
+
+	if (version != NULL) {
+		version->api    = sys_kernel_version_get();
+		version->kernel = sys_kernel_version_get();
+	}
+
+	sprintf(os_str, "Zephyr V%2d.%2d.%2d",
+		SYS_KERNEL_VER_MAJOR(version->kernel),
+		SYS_KERNEL_VER_MINOR(version->kernel),
+		SYS_KERNEL_VER_PATCHLEVEL(version->kernel));
+
+	if ((id_buf != NULL) && (id_size > strlen(os_str))) {
+		memcpy(id_buf, os_str, strlen(os_str)+1);
+	}
+
+	return (osOK);
+}
+
+/**
+ * @brief Lock the RTOS Kernel scheduler.
+ */
+int32_t osKernelLock(void)
+{
+	int temp = _current->base.sched_locked;
```

Comment:
This API cannot be called from ISR context. I think we need a check something like this here:

```
if (k_is_in_isr()) {
        return osErrorISR;
    }
```

Reply:
addressed. Thanks for pointing out.